src/test/java/domain/profile/CustomerProfileServiceTest.java

```diff
@@ -0,0 +1,606 @@
+        assertEquals(customerProfile.getEmail(), result.deletedCustomerProfile().get().getEmail());
+
+    }
+    @Test
+    public void starListing_WhenCustomerNotFound_SetCustomerNotFoundToTrue(){
+        StarListing starListing = StarListing.builder()
+                .customerId("customerId")
+                .listingId("listingId")
+                .build();
+        when(customerProfileRepository.getCustomerProfile(starListing.getCustomerId())).thenReturn(Optional.empty());
+
+        CustomerStarResult result = customerProfileService.starListing(starListing);
+
+        assertTrue(result.isCustomerNotFound());
+        assertFalse(result.isListingNotFound());
+        assertFalse(result.getCustomerProfile().isPresent());
+    }
+    @Test
+    public void starListing_WhenListingNotFound_SetListingNotFoundToTrue(){
+        StarListing starListing = StarListing.builder()
+                .customerId("customerId")
+                .listingId("listingId")
+                .build();
+        when(customerProfileRepository.getCustomerProfile(starListing.getCustomerId())).thenReturn(Optional.of(new CustomerProfileEntity()));
+        when(listingRepository.getListing(starListing.getListingId())).thenReturn(Optional.empty());
+
+        CustomerStarResult result = customerProfileService.starListing(starListing);
+
+        assertFalse(result.isCustomerNotFound());
+        assertTrue(result.isListingNotFound());
+        assertFalse(result.getCustomerProfile().isPresent());
+    }
+
+    @Test
+    public void starListing_WhenListingAndCustomerExist_ReturnCustomerStarListingResult(){
+        StarListing starListing = StarListing.builder()
+                .customerId("customerId")
+                .listingId("listingId")
+                .build();
+        CustomerProfileEntity customerProfileEntity = CustomerProfileEntity.builder()
+                .id("profileId")
+                .email("tester@gmail.com")
+                .password("1234")
+                .name("tester")
+                .avatar("123")
+                .phone("123456")
+                .longitude(0.0)
+                .latitude(0.0)
+                .postedListings(List.of())
+                .starredListings(new ArrayList<>())
+                .build();
+
+        ListingEntity listingEntity_1 = Mockito.mock(ListingEntity.class);
+        when(listingEntity_1.getId()).thenReturn("listingId");
+
+        when(customerProfileRepository.getCustomerProfile(starListing.getCustomerId())).thenReturn(Optional.of(customerProfileEntity));
+        when(listingRepository.getListing(starListing.getListingId())).thenReturn(Optional.of(listingEntity_1));
+
+
+        CustomerStarResult result = customerProfileService.starListing(starListing);
+
+        assertFalse(result.isCustomerNotFound());
+        assertFalse(result.isListingNotFound());
+        result.getCustomerProfile().ifPresent(customerProfile -> {
```

Comment:
shouldn't have if statement as it indicates different path for testing.
Instead, assert the result you're expecting